CDTDatastore/query/CDTQIndex.m

```diff
@@ -52,22 +52,25 @@ + (NSArray *)validSettings
 
 - (instancetype)initWithFields:(NSArray *)fieldNames
                      indexName:(NSString *)indexName
-                     indexType:(NSString *)indexType
+                     indexType:(CDTQIndexType)indexType
                  indexSettings:(NSDictionary *)indexSettings
 {
     self = [super init];
     if (self) {
         _fieldNames = fieldNames;
         _indexName = indexName;
-        _indexType = indexType;
+        _indexType = [CDTQIndexManager stringForIndexType:indexType];
+        _type = indexType;
```

Comment:
No need for assigning `_type` any more (might not work anyway if you have getter/setter for `type`).